src/node/service.rs

```diff
@@ -1,158 +1,109 @@
-      Err(error) => Err(error),
-    }
-  }
-
-  // ===========================================================================
-  // Block Management
-  // ===========================================================================
-
-  /// Update the prioritization of blocks to check.
-  ///
-  /// Note: The results of all checks will be sent
-  ///       as BlockValid and BlockInvalid updates.
-  pub fn check_blocks(&self, priority: Vec<BlockId>) -> Result<(), Error> {
-    self.inner.borrow_mut().check_blocks(priority)
-  }
-
-  /// Update the block that should be committed.
-  ///
-  /// Note: This block must already have been checked.
-  pub fn commit_block(&self, block_id: BlockId) -> Result<(), Error> {
-    self.inner.borrow_mut().commit_block(block_id)
-  }
+impl Deref for PowService {
+  type Target = Box<dyn Service>;
 
-  /// Signal that this block is no longer being committed.
-  pub fn ignore_block(&self, block_id: BlockId) -> Result<(), Error> {
-    self.inner.borrow_mut().ignore_block(block_id)
-  }
-
-  /// Mark this block as invalid from the perspective of consensus.
-  ///
-  /// Note: This will also fail all descendants.
-  pub fn fail_block(&self, block_id: BlockId) -> Result<(), Error> {
-    self.inner.borrow_mut().fail_block(block_id)
-  }
-
-  // ===========================================================================
-  // P2P
-  // ===========================================================================
-
-  /// Send a consensus message to a specific, connected peer.
-  #[allow(clippy::ptr_arg)]
-  pub fn send_to(&self, peer: &PeerId, type_: &str, payload: Vec<u8>) -> Result<(), Error> {
-    self.inner.borrow_mut().send_to(peer, type_, payload)
+  fn deref(&self) -> &Self::Target {
+    &(self.service)
   }
+}
 
-  /// Broadcast a message to all connected peers.
-  pub fn broadcast(&self, type_: &str, payload: Vec<u8>) -> Result<(), Error> {
-    self.inner.borrow_mut().broadcast(type_, payload)
+impl DerefMut for PowService {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    &mut (self.service)
   }
+}
 
-  // ===========================================================================
-  // Querying
-  // ===========================================================================
-
-  /// Retrieve consensus-related information about blocks
-  pub fn get_blocks(&self, block_ids: Vec<BlockId>) -> Result<HashMap<BlockId, Block>, Error> {
-    self.inner.borrow_mut().get_blocks(block_ids)
+impl PowService {
+  pub fn new(service: Box<dyn Service>) -> Self {
+    PowService { service }
   }
 
-  /// Retrieve consensus-related information about a specific block
-  pub fn get_block(&self, block_id: &[u8]) -> anyhow::Result<Block> {
+  pub fn get_block(&mut self, block_id: &[u8]) -> Result<Block, Error> {
     self
-      .get_blocks(vec![block_id.to_owned()])?
+      .service
+      .get_blocks(vec![block_id.to_owned()])
+      .expect(&format!("Block {}", dbg_hex!(block_id)))
```

Comment:
Do we want to panic in this case? Why not just log it and return the error early?

Reply:
On further reading, I get what you mean now. I agree.